cmd/set.go

```diff
@@ -69,22 +87,24 @@ Sets the value of 'endpoint' equal to 'east-db' in the 'common' config (the defa
 $ fab set --subcomponent "myapp.mysubcomponent" data.replicas=5 
 
 Sets the subkey "replicas" in the key 'data' equal to 5 in the 'common' config (the default) for the subcomponent 'mysubcomponent' of the subcomponent 'myapp'.
+
+$ fab set --subcomponent "myapp.mysubcomponent" data.replicas=5 --newconfigfail
+
+Uses the --newconfigfail switch to fail the action if a new config pair is created versus updating an existing value.
 `,
 	RunE: func(cmd *cobra.Command, args []string) error {
 		if len(args) < 1 {
 			return errors.New("'set' takes one or more key=value arguments")
 		}
 
-		subcomponent := cmd.Flag("subcomponent").Value.String()
-		environment := cmd.Flag("environment").Value.String()
-
-		return Set(environment, subcomponent, args)
+		return Set(environment, subcomponent, args, newconfigfail)
 	},
 }
 
 func init() {
-	setCmd.PersistentFlags().String("environment", "common", "Environment this configuration should apply to")
-	setCmd.PersistentFlags().String("subcomponent", "", "Subcomponent this configuration should apply to")
+	setCmd.PersistentFlags().StringVar(&environment, "environment", "common", "Environment this configuration should apply to")
+	setCmd.PersistentFlags().StringVar(&subcomponent, "subcomponent", "", "Subcomponent this configuration should apply to")
+	setCmd.PersistentFlags().BoolVar(&newconfigfail, "newconfigfail", false, "Fail the task if a new config pair is created versus updating an existing value")
```

Comment:
Let's call this flag `no-new-config-keys` where it 'Prevent creation of new config keys and only allow updating existing config values.'